### PR TITLE
Creating a simpler script.

### PR DESCRIPTION
**Pull request by gvwilson**
This creates a simpler script `github-push-checklist.py` that creates issues from a flat text file, one issue per line.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,4 @@
+*~
```
Comment (wking) [inline .gitignore:1]: Ignoring your editor's temporary files is more of a per-user issue than something specific to this repository.  I'd recommend you include this in your per-user `~/.gitignore`.

**Comment by wking** [comment]
On Mon, Apr 13, 2015 at 01:00:21PM -0700, Greg Wilson wrote:

> This creates a simpler script `github-push-checklist.py` that
> creates issues from a flat text file, one issue per line.

We talked about this approach abit offline, and I still think it's
more useful to support issue bodies and grouping milestones.  Quoting
here, I said [1]:

> I'm concerned that folks _won't_ use a script that throws up around
> twenty unstructured issues with vague or complicated summaries
> (depending on how much context you want to cram into your single line).
> 
> Are you concerned about script-development time?  Or the complexity of
> maintaining the checklist filesystem?  In both cases, I think my
> proposal is a slightly higher bar.  But I also think the increased
> cost on the dev/maintainer side is outweighed by the increased
> usability on the lead-instructor side (and workshops happen much more
> frequently than the script or checklist filesytem would need to be
> updated).

My current implementation is past the script-development phase, so the
only outstanding costs I see are script maintenance (which I expect to
be negligable, and am happy to cover on my own) and checklist-template
maintenance.  Maintaining checklist templates in my directory-tree
format is a bit more work per-item than your flat-file format, but I
think the usibility of issue bodies and optional grouping milestones
is worth it.  If that's not convincing, can you go into more detail
about why you expect my directory-tree format will be difficult to
maintain, or why you think issue bodies and/or milestones aren't
useful?

[1]: Subject: Re: a small script to populate workshop repos with issues
     Date: Sat, 11 Apr 2015 13:39:12 -0700
     Message-ID: 20150411203912.GC10161@odin.tremily.us

**Comment by gvwilson** [comment]
On 2015-04-13 4:20 PM, W. Trevor King wrote:

> On Mon, Apr 13, 2015 at 01:00:21PM -0700, Greg Wilson wrote:
> 
> > This creates a simpler script `github-push-checklist.py` that
> > creates issues from a flat text file, one issue per line.
> 
> We talked about this approach abit offline, and I still think it's
> more useful to support issue bodies and grouping milestones.

I disagree, so I thought we could make both available and people could 
use whichever appealed to them most.

**Comment by wking** [comment]
On Mon, Apr 13, 2015 at 01:34:44PM -0700, Greg Wilson wrote:

> 2015-04-13 4:20 PM, W. Trevor King:
> 
> > Mon, Apr 13, 2015 at 01:00:21PM -0700, Greg Wilson:
> > 
> > > This creates a simpler script `github-push-checklist.py` that
> > > creates issues from a flat text file, one issue per line.
> > 
> > We talked about this approach abit offline, and I still think it's
> > more useful to support issue bodies and grouping milestones.
> 
> I disagree, so I thought we could make both available and people
> could use whichever appealed to them most.

I don't like maintaining code when I don't understand the use-case
behind it, so I'd rather not merge an alternative script here.  But if
the template format I prefer doesn't seem workable to you, I'm fine
with you forking the project and pushing the flat-file version to PyPI
under a different name (since my code is under the BSD-2 after all).

**Comment by gvwilson** [comment]
On 2015-04-13 4:42 PM, W. Trevor King wrote:

> On Mon, Apr 13, 2015 at 01:34:44PM -0700, Greg Wilson wrote:
> 
> > On 2015-04-13 4:20 PM, W. Trevor King wrote:
> > 
> > > On Mon, Apr 13, 2015 at 01:00:21PM -0700, Greg Wilson wrote:
> > > 
> > > > This creates a simpler script `github-push-checklist.py` that
> > > > creates issues from a flat text file, one issue per line.
> > > 
> > > We talked about this approach abit offline, and I still think it's
> > > more useful to support issue bodies and grouping milestones.
> > 
> > I disagree, so I thought we could make both available and people
> > could use whichever appealed to them most.
> 
> I don't like maintaining code when I don't understand the use-case
> behind it, so I'd rather not merge an alternative script here. But if
> the template format I prefer doesn't seem workable to you, I'm fine
> with you forking the project and pushing the flat-file version to PyPI
> under a different name (since my code is under the BSD-2 after all).
> Thanks - I'll do that, then we can advertise both.  I've been wrong many 
> times before...
> Thanks for building this,
> Greg

## 

Email longa, vita brevis.

**Comment by wking** [comment]
On Mon, Apr 13, 2015 at 01:40:18PM -0700, W. Trevor King wrote:

> I don't like maintaining code when I don't understand the use-case
> behind it…

I'm also open to understanding the motivation behind the flat-file
approach.  “Lets just put it out there and see which one gets used” is
fine, but my guess is that:
- Folks using the script don't really care about the template format.
  They just call the script and get the issues.  So choosing the
  initial format is really a question for the template maintainer.
- As a template maintainer, I expect you'd choose to use your
  flat-file version, so perhaps you could share a few words about why
  you prefer that choice?
- How likely script users are to be repeat users depends on how useful
  they find the auto-created issues.  I think issues with descriptive
  bodies which are optionally grouped by milestones would be more
  useful here (clearly, since they're a superset of unsorted, bodyless
  issues), so I think my format is the way to go.

But again, you're the checklist template maintainer, so you're free to
pick whatever format you like best without explaining your choices to
me, if that makes the most sense to you.

**Comment by wking** [comment]
On Mon, Apr 13, 2015 at 01:58:28PM -0700, Greg Wilson wrote:

> Thanks for building this,

No problem.  I'm always happy to throw together some tooling to make
SWC run more smoothly :).

**Pull request closed without merging**